test/extended/dr/static_pod_rollouts.go

```diff
@@ -0,0 +1,139 @@
+package dr
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	librarygov1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Suite:openshift/etcd/recovery][Disruptive] etcd", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("etcd-static-pod-rollouts").AsAdmin()
+
+	etcdNamespace := "openshift-etcd"
+	var initialEtcdPodCount int   // the number of etcd pods in the initial state of the cluster
+	var etcdTargetNodeName string // node from which the etcd pod is to be removed
+	var initialLogLevel string
+
+	g.BeforeEach(func() {
+		isSingleNode, err := exutil.IsSingleNode(context.Background(), oc.AdminConfigClient())
+		o.Expect(err).ToNot(o.HaveOccurred())
+		if isSingleNode {
+			g.Skip("the test is for etcd peer communication which is not valid for single node")
+		}
+
+		etcdCluster, err := oc.AdminOperatorClient().OperatorV1().Etcds().Get(context.Background(), "cluster", metav1.GetOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+		initialLogLevel = string(etcdCluster.Spec.LogLevel)
+		o.Expect(initialLogLevel).ToNot(o.Equal("Debug"), "log level is not different from Debug, which is the change being made to trigger a rollout")
+
+		isUnsupportedUnsafeEtcd, err := isUnsupportedUnsafeEtcd(&etcdCluster.Spec.StaticPodOperatorSpec)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		o.Expect(isUnsupportedUnsafeEtcd).To(o.BeFalse(), "quorum guard is off, this will allow the CEO to act without quorum")
```

Comment:
This error message reads a little ambiguous on what we're asserting and why it failed.
So we're asserting that the quorum guard must be on then:
```suggestion
		o.Expect(isUnsupportedUnsafeEtcd).To(o.BeFalse(), "quorum guard must not be off, expected useUnsupportedUnsafeNonHANonProductionUnstableEtcd: false")
```

```
{"spec": {"unsupportedConfigOverrides": {"useUnsupportedUnsafeNonHANonProductionUnstableEtcd": true}}}
```